webrender/src/display_list_flattener.rs

```diff
@@ -169,6 +169,9 @@ pub struct DisplayListFlattener<'a> {
     /// The map of all font instances.
     font_instances: FontInstanceMap,
 
+    /// The map keeping track of the root clip chains associated with pipelines.
+    pipeline_clip_chains: FastHashMap<PipelineId, ClipChainIndex>,
```

Comment:
If you want to avoid the cost of hashing here, this can just be a stack. You would push ClipChainIndex(0) for the root pipeline and then push again when processing an iframe element.

Reply:
good point!